Hybrid/example.rb

```diff
@@ -0,0 +1,270 @@
+require 'azure_mgmt_storage'
+require 'azure_mgmt_compute'
+
+LOCAL = 'local'
+GROUP_NAME = 'azurestack-sample-compute'
+OpenSSL::SSL::VERIFY_PEER = OpenSSL::SSL::VERIFY_NONE 
+Storage = Azure::Storage::Profiles::V2017_03_09::Mgmt
+Network = Azure::Network::Profiles::V2017_03_09::Mgmt
+Compute = Azure::Compute::Profiles::V2017_03_09::Mgmt
+Resources = Azure::Resources::Profiles::V2017_03_09::Mgmt
+
+StorageModels = Storage::Models
+NetworkModels = Network::Models
+ComputeModels = Compute::Models
+ResourceModels = Resources::Models
+
+# This sample shows how to manage a Azure virtual machines using using the Azure Resource Manager APIs for Ruby.
+#
+# This script expects that the following environment vars are set:
+#
+# AZURE_TENANT_ID: with your Azure Active Directory tenant id or domain
+# AZURE_CLIENT_ID: with your Azure Active Directory Application Client ID
+# AZURE_CLIENT_SECRET: with your Azure Active Directory Application Secret
+# AZURE_SUBSCRIPTION_ID: with your Azure Subscription Id
+#
+def run_example
+  #
+  # Create the Resource Manager Client with an Application (service principal) token provider
+  #
+  subscription_id = ENV['AZURE_SUBSCRIPTION_ID'] || '11111111-1111-1111-1111-111111111111' # your Azure Subscription Id
+  active_directory_settings = get_active_directory_settings()
+
+
+  provider = MsRestAzure::ApplicationTokenProvider.new(
+      ENV['AZURE_TENANT_ID'],
+      ENV['AZURE_CLIENT_ID'],
+      ENV['AZURE_CLIENT_SECRET'],
+      active_directory_settings
+      )
+  credentials = MsRest::TokenCredentials.new(provider)
+
+  options = {
+      credentials: credentials,
+      subscription_id: subscription_id,
+      active_directory_settings: active_directory_settings,
+      base_url: 'https://management.local.azurestack.external' # Azure Resource Manager Url
+  }
+
+  resource_client = Azure::Resources::Profiles::V2017_03_09::Mgmt::Client.new(options)
+  network_client = Azure::Network::Profiles::V2017_03_09::Mgmt::Client.new(options)
+  storage_client = Azure::Storage::Profiles::V2017_03_09::Mgmt::Client.new(options)
+  compute_client = Azure::Compute::Profiles::V2017_03_09::Mgmt::Client.new(options)
+
+  #
+  # Managing resource groups
+  #
+  resource_group_params = ResourceModels::ResourceGroup.new.tap do |rg|
+    rg.location = LOCAL
+  end
+
+  # Create Resource group
+  puts 'Create Resource Group'
+  print_group resource_client.resource_groups.create_or_update(GROUP_NAME, resource_group_params)
+
+  postfix = rand(1000)
+  storage_account_name = "rubystor#{postfix}"
+  puts "Creating a standard storage account named #{storage_account_name} in resource group #{GROUP_NAME}"
+  storage_create_params = StorageModels::StorageAccountCreateParameters.new.tap do |account|
+    account.location = LOCAL
+    account.sku = StorageModels::Sku.new.tap do |sku|
+      sku.name = StorageModels::SkuName::StandardLRS
+      sku.tier = StorageModels::SkuTier::Standard
+    end
+    account.kind = StorageModels::Kind::Storage
```

Comment:
can you add a comment here on how this SA creation differs from azure.  we support encryption settings only on blob

Reply:
Comment added